File: lab4/Entities/ApprovedActivity.cs
using System;
using System.Collections.Generic;

namespace lab4.Entities
{
    public class ApprovedActivityStatsJson : EntityJson
    {
        public string projectId { get; set; } = "";
        public int time { get; set; }
    }

    public class ApprovedActivityJson : EntityJson
    {
        public DateTime month { get; set; } = new DateTime();
        public string userName { get; set; } = "";
        public string projectId { get; set; } = "";
        public int time { get; set; }
    }

    public class ApprovedActivity : IEntity<ApprovedActivityJson>
    {
        // PK
        public DateTime Month { get; set; } = new DateTime();
        public string UserName { get; set; } = "";
        public string ProjectId { get; set; } = "";

        public int Time { get; set; }
        
        // Parents
        public virtual Project? Project { get; set; }
        public virtual UserMonth? UserMonth { get; set; }

        public ApprovedActivityJson toJSON()
        {
            return new ApprovedActivityJson
            {
                month=Month,
                userName=UserName,
                projectId=ProjectId,
                time=Time
            };
        }
    }
}

File: lab4/ClientApp/src/entities/ApprovedActivity.ts
export class ApprovedActivity {
    month: Date;
    userName: string;
    projectId: string;
    time: number;

    constructor(month: Date, userName: string, projectId: string, time: number) {
        this.month = month;
        this.userName = userName;
        this.projectId = projectId;
        this.time = time;
    }

    toJSON() {
        return {
            month: this.month,
            userName: this.userName,
            projectId: this.projectId,
            time: this.time
        };
    }

    static fromJSON(json: any) {
        return new ApprovedActivity(json['month'], json['userName'], json['projectId'], json['time']);
    }

    static createEmpty() {
        return new ApprovedActivity(new Date(), '', '', 0);
    }
};

export default ApprovedActivity;
